native/packages/react-native-bpk-component-button/src/BpkButton.js

```diff
@@ -177,76 +69,39 @@ const BpkButton = (props) => {
     }
   }
 
-  const accessibilityTraits = ['button'];
-  if (disabled) {
-    accessibilityTraits.push('disabled');
-  }
-
-  const textStyle = [
-    getStyleForElement('text', props),
-    getThemingForElement('text', theme, props),
-  ];
-
   const renderIcon = () => {
     if (!icon) {
       return null;
     }
     if (typeof icon === 'string') {
-      return <BpkIcon icon={icon} style={textStyle} small={!large} />;
+      return <BpkIcon icon={icon} style={textStyle(theme, props)} small={!large} />;
     }
     return icon;
   };
 
-  const getIOSContainer = buttonContent => (<LinearGradient
-    colors={getGradientColors(theme, props)}
-    style={[getStyleForElement('container', props), style]}
-  >
-    <TouchableHighlight
-      style={[getStyleForElement('button', props), getThemingForElement('button', theme, props)]}
-      disabled={disabled}
+  return (
+    <BpkButtonContainer
+      {...props}
```

Comment:
Usually when we destructure `props` and spread the remainder to `rest` we are explicit about what gets applied to underlying component. 

I can see some of the appeal of just spreading `props` onto the component but it's a bit of a sledgehammer approach - i.e. not very clear or explicit. It's asking for (unknown prop type warnings)

Also when ever I see a variable being destructured I don't expect it to ever be referenced again - is this just me?